substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Distribution pallet.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// Re-export all pallet parts, this is needed to properly import the pallet into the runtime.
+pub use pallet::*;
+pub mod functions;
+mod types;
+pub use pallet_distribution as Distribution;
+pub use types::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+pub mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + Distribution::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The minimum duration for which votes are locked
+		#[pallet::constant]
+		type VoteLockingPeriod: Get<BlockNumberFor<Self>>;
+
+		/// The maximum number of whitelisted projects per nomination round
+		#[pallet::constant]
+		type MaxWhitelistedProjects: Get<u32>;
+
+		/// Time during which it is possible to cast a vote or change an existing vote.
+		///
+		#[pallet::constant]
+		type VotingPeriod: Get<BlockNumberFor<Self>>;
+
+		/// Used for Pallet testing only. Represents the Total Reward distributed
+		#[pallet::constant]
+		type TemporaryRewards: Get<BalanceOf<Self>>;
```

Comment:
why is this needed?
if it is truly unavoidable (which I don't think it is the case here), you should put it under a feature flag. e.g.
https://github.com/paritytech/polkadot-sdk/blob/0cd577ba1c4995500eb3ed10330d93402177a53b/bridges/snowbridge/pallets/inbound-queue/src/lib.rs#L129-L130